### PR TITLE
TILA-247 reservation unit search enhancements

### PR DESCRIPTION
**Pull request by kieferbonk**

### CODE REVIEW

File: src/application/page1/ReservationUnitModal.tsx [hunk 1]
```diff
@@ -125,63 +132,156 @@ const Text = styled.span`
...
+          options={reservationUnitTypeOptions}
+          label={t('ReservationUnitModal.searchReservationUnitTypeLabel')}
+          onChange={(selection: OptionType): void => {
+            setReservationUnitType(selection);
+          }}
+          defaultValue={emptyOption}
+        />
+      </Filters>
+      <ButtonContainer>
+        <SearchButton onClick={() => searchResults()}>
```
Comment (siren) [inline src/application/page1/ReservationUnitModal.tsx:262]: Looks like this button now tries to submit the form behind the modal tool, need to add call to "e.preventDefault()' ?

**Review by siren — COMMENTED** [review]
There's something strange happening with the Spinner, sometimes I see the following error when clicking 'Hae'

```
LoadingSpinner-7786d9a3.js:1 Uncaught TypeError: Cannot read property 'removeChild' of null
    at LoadingSpinner-7786d9a3.js:1
```

File: src/application/page1/ReservationUnitModal.tsx [hunk 2]
```diff
@@ -125,63 +132,156 @@ const Text = styled.span`
...
 
-  const results = useAsync(async () => {
-    if (q === null) {
-      return [];
-    }
-    return getReservationUnits({ search: q });
-  }, [q]);
+  const searchResults = async () => {
+    setSearching(true);
+    const searchCriteria = {
```
Comment (siren) [inline src/application/page1/ReservationUnitModal.tsx:208]: What about the ApplicationPeriod filtering?

Comment (kieferbonk) [reply src/application/page1/ReservationUnitModal.tsx:208]: Forgot about that - added.

File: src/application/page1/ReservationUnitModal.tsx (hunk 1 above)
Comment (kieferbonk) [reply src/application/page1/ReservationUnitModal.tsx]: Yes, good catch. Fixed.

**Comment by kieferbonk** [comment]
I was unable to reproduce the spinner problem. Updated hds to new version, that adressed some spinner bug.

File: src/application/page1/ReservationUnitModal.tsx (hunk 2 above)
Comment (siren) [reply src/application/page1/ReservationUnitModal.tsx:208]: Not super important right now, but the api also supports filtering with parameter application_period , for example application_period=1

**Review by siren — APPROVED** [review]

**Review by siren — APPROVED** [review]

**Pull request merged**